driver/src/main/java/org/neo4j/driver/internal/connector/socket/TrustOnFirstUseTrustManager.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal.connector.socket;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.X509TrustManager;
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * References:
+ * http://stackoverflow.com/questions/6802421/how-to-compare-distinct-implementations-of-java-security-cert-x509certificate?answertab=votes#tab-top
+ */
+
+class TrustOnFirstUseTrustManager implements X509TrustManager
+{
+    /**
+     * A list of pairs (known_server_ip, certificate) are stored in this file.
+     * When establishing a SSL connection to a new server, we will save the server's IP and its certificate in this
+     * file.
+     * Then when we try to connect to a known server again, we will authenticate the server by checking if it provides
+     * the same certificate as the one saved in this file.
+     */
+    static final String KNOWN_CERTS_FILE_PATH = "known_certs";
```

Comment:
+1 Great suggestions. 

Is it possible for a client to connect using TLS _without_ having a writable filesystem available?